src/engine/Server.cpp

```diff
@@ -163,47 +164,70 @@ void Server::run(const string& indexBaseName, bool useText, bool usePatterns,
 }
 
 // _____________________________________________________________________________
-ad_utility::UrlParser::UrlPathAndParameters Server::getUrlPathAndParameters(
+ad_utility::UrlParser::ParsedHTTPRequest Server::getUrlPathAndParameters(
     const ad_utility::httpUtils::HttpRequest auto& request) {
+  // For an HTTP request, `request.target()` yields the HTTP Request-URI.
+  // This is a concatenation of the URL path and the query strings.
+  ad_utility::UrlParser::ParsedHTTPRequest parsedUrl =
+      ad_utility::UrlParser::parseRequestTarget(request.target());
+
   if (request.method() == http::verb::get) {
-    // For a GET request, `request.target()` yields the part after the domain,
-    // which is a concatenation of the path and the query string (the query
-    // string starting with "?").
-    return ad_utility::UrlParser::parseGetRequestTarget(request.target());
+    // There can be valid GET requests without a query. E.g. the command to get
+    // index statistics.
+    if (parsedUrl.parameters_.contains("query")) {
+      parsedUrl.query_ = parsedUrl.parameters_["query"];
+    }
```

Comment:
Are the custom QLever commands only valid for `GET` requests or can they also happen for `POST`?

Reply:
I see no reason why we shouldn't have the same behavior for GET and POST in this respect. 
The only difference seems to be the way that the query is specified.